extension/src/project/readOnlyDocument.ts

```diff
@@ -97,12 +97,35 @@ export class ReadonlyDocument implements vscode.TextDocument {
             this.lines = fileContent.split('\r\n');
             this.lineCount = this.lines.length;
         }
-        this.atomsForest = [];
-        if (languageId === 'autolisp'){
+    }
+
+    
+    // Converted this from a constant data feature into an on-demand feature that once used is essentially cached for future queries.
+    get atomsForest(): Array<string|Sexpression> {
+        if (this.languageId === 'autolisp') {
+            if (this._atomsForest && this._atomsForest.length > 0){
```

Comment:
Is this field initialized as null or undefined? Considering that the user is possible to open an empty LSP file, what's the length of this array in that case? if it's greater than 0, I'm then OK with the second condition, otherwise I think we shouldn't check the length here, because it will update the cache for every call.